### DISCUSSION TITLE
Loading GoRules Engine in NextJS

### DISCUSSION THREAD

**Discussion by john--kane** [vercel/next.js#54170] [Help]
### Summary

I'm trying to load the gorules engine in NextJS but I'm getting an error. Is there a work around or something I'm missing to make this work? 

I'm using App Router 

"dependencies": {
    "@gorules/zen-engine": "^0.6.0",
    "@radix-ui/react-label": "^2.0.2",
    "@radix-ui/react-slot": "^1.0.2",
    "@types/node": "20.5.0",
    "@types/react": "18.2.20",
    "@types/react-dom": "18.2.7",
    "autoprefixer": "10.4.15",
    "class-variance-authority": "^0.7.0",
    "clsx": "^2.0.0",
    "eslint": "8.47.0",
    "eslint-config-next": "13.4.16",
    "lucide-react": "^0.268.0",
    "next": "13.4.16",
    "postcss": "8.4.27",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-dropzone": "^14.2.3",
    "react-hook-form": "^7.45.4",
    "tailwind-merge": "^1.14.0",
    "tailwindcss": "3.3.3",
    "tailwindcss-animate": "^1.0.6",
    "typescript": "5.1.6"
  }

### Additional information

```js
./node_modules/.pnpm/@gorules+zen-engine-darwin-x64@0.6.0/node_modules/@gorules/zen-engine-darwin-x64/zen-engine.darwin-x64.node
Module parse failed: Unexpected character '�' (1:0)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
(Source code omitted for this binary file)
```

**Comment by diazjavier** [comment]
I have the same issue. Any solution over there?

**Comment by supermacro** [comment]
> `zen-engine-darwin-x64`

Looks like you may be trying to run a nodejs module in a browser environment. Have you correctly configured zen engine to be run in a server only context?
